pimd/pim6_sock.c

```diff
@@ -0,0 +1,335 @@
+/*
+ * PIM for FRR
+ * Copyright (C) 2022 Dell Technologies
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include <zebra.h>
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netinet/igmp.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+#include <netdb.h>
+#include <errno.h>
+
+#include "log.h"
+#include "privs.h"
+#include "if.h"
+#include "vrf.h"
+#include "sockopt.h"
+#include "lib_errors.h"
+
+#include "pimd.h"
+#include "pim_mroute.h"
+#include "pim_sock.h"
+#include "pim_str.h"
+#include "pim_addr.h"
+
+/* GLOBAL VARS */
+
+int pim_socket_raw(int protocol)
```

Comment:
Yeah I am totally not a fan of cut-n-pasting code and having 2 identical function names that do the exact same thing in different files.  Please go through and abstract as appropriate.

Reply:
Ack